src/db.rs

```diff
@@ -626,6 +672,22 @@ impl Db {
         self.inner.delete_with_options(key, options).await;
     }
 
+    /// Write a batch of put/delete operations atomically to the database. All batch
+    /// operations are guaranteed to end up in the same WAL SST (or L0 memtable if
+    /// WAL is disabled).
+    pub async fn write_batch(&self, batch: &WriteBatch) {
```

Comment:
Maybe we can introduce a new background tokio task to prevent the batch from being too large and blocking other operations? If it cannot be completed in this PR, I can help after completing the cache admission policy.
The following are reference material.
[WickDB writeBarch impl](https://github.com/Fullstop000/wickdb/blob/master/src/db/mod.rs#L836)

Reply:
i just recalled that i had a note about the write path of leveldb and translated it into english https://flaneur2020.github.io/posts/2021-04-11-leveldb-writes/ (hopefully the LLM did not hallucinate...)

Reply:
> Maybe we can introduce a new background tokio task to prevent the batch from being too large and blocking other operations

Good idea. I'm realizing now that doing so would require us to clone the `&[u8]` (which I introduced in https://github.com/slatedb/slatedb/commit/cb13d84c5f34d77a86bd870838a410e0c60d30aa) so that the `WriteBatch` could be moved into the `spawn_blocking` closure. The clone could be expensive, though. I suppose that would be an argument for the AsRef approach you suggested, though I still don't like what it does to the API. Alternatively, we could go back to the original owned `Vec[u8]` style, which would really just push the issue upstream to the client. 🤔

Reply:
I guess it looks fine for me to pass owned Vec<u8> in the put() calls. It helps reduce copying, and it's not a burden for users to use this API."

"however i notice rocksdb-rs indeed used AsRef<u8> in the put() https://rust.velas.com/rocksdb/struct.WriteBatch.html#method.put, i'm also curious about how they did this. @KaguraMilet wanna here from your further suggestions."

Reply:
A simple prototype
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5e151e689d0058a28733d3edcd696350)

Reply:
I think it is harmless and there is no need to conduct too deep research on it.

Reply:
> Maybe we can introduce a new background tokio task to prevent the batch from being too large and blocking other operations?

How will this help? The background task will still need to lock the table so that it can apply the whole batch atomically right? I suppose the benefit is that batch writes can proceed without waiting on lock-holders to write the batch into the table, but only up to a point - eventually they'll get blocked on the channel being full.

Reply:
Oh I see what you mean now - we don't want to block the tokio runtime threads on this large/contentious critical section. That makes sense. Though now the batch writes might get blocked on whatever synchronous i/o might be happening on the background threads. Seems better to use an async lock instead.

Reply:
i suppose the benefits in the leveldb's queuing approach is that it may help aggregate the writes from multiple threads into a single IO, thus making small writes much more efficient to batch them and let them bigger than a 4~8kb fs page.

yes, it's worth to think about whether the benefits is still hold true when the underlying storage is object store.

Reply:
Gotcha. I think we already get that benefit by virtue of the writes accumulating into an in-memory data structure (the wal table, or if wal is disabled the memtable) that is then written to object store (periodically from the wal flush task in the case of wal, and at a size threshold by the memtable flush task in the case the wal is off).

Reply:
https://github.com/cockroachdb/pebble/blob/master/docs/rocksdb.md#commit-pipeline 

i feel this link which @criccomini shared is worth to read, it discuesses why pebble did not follow the leveldb/rocksdb's approach and the design consideration of the commit pipeline, there might have some more considerations other than batching IO. (but i havent got time to read yet, hopefully could have time to read it tonight)

Reply:
My latest PR has a commit pipeline (`batch_write.rs`). PTAL.